src/server/api/models/docgen/service.js

```diff
@@ -0,0 +1,184 @@
+const fs = require('fs')
+const p = require('path')
+const docgen = require('react-docgen')
+
+const utils = require('react-docgen/dist/utils')
+const resolveHOC = require('react-docgen/dist/utils/resolveHOC.js').default
+
+const isComponentDefinition = (path) => {
+  return utils.isReactCreateClassCall(path)
+    || utils.isReactComponentClass(path)
+    || utils.isStatelessComponent(path)
+}
+
+const resolveDefinition = (definition, types) => {
+  if (utils.isReactCreateClassCall(definition)) {
+    const resolvedPath = utils.resolveToValue(definition.get('arguments', 0))
+    if (types.ObjectExpression.check(resolvedPath.node)) {
+      return resolvedPath
+    }
+  } else if (utils.isReactComponentClass(definition)) {
+    utils.normalizeClassDefinition(definition)
+    return definition
+  } else if (utils.isStatelessComponent(definition)) {
+    return definition
+  }
+  return null
+}
+
+const componentResolver = (ast, recast) => {
+  const types = recast.types.namedTypes
+  const components = []
+
+  const exportDeclaration = (path) => {
+    const definitions = utils.resolveExportDeclaration(path, types)
+      .reduce((acc, definition) => {
+        if (isComponentDefinition(definition)) {
+          acc.push(definition)
+        } else {
+          const resolved = utils.resolveToValue(resolveHOC(definition))
+          if (resolved.node.type === 'ImportDeclaration') {
+            // Manage imported components in HOC from different file
```

Comment:
Why do you pass by an exception instead of recalling `exportDeclaration(newPath)`